### PR TITLE
A tricky way to modify the engine interface

### PR DESCRIPTION
**Pull request by nnsgmsone**
**What type of PR is this?**

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

issue #3591

**What this PR does / why we need it:**

Not Available

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
My eyes!    So all it does, is to add a ctx parameter, is this true?

File: pkg/vm/engine/types.go [hunk 1]
```diff
@@ -15,46 +15,61 @@
...
+// Attribute is a column
 type Attribute struct {
-	Name    string      // name of attribute
-	Alg     compress.T  // compression algorithm
-	Type    types.Type  // type of attribute
-	Default DefaultExpr // default value of this attribute.
-	Primary bool        // if true, it is primary key
-	Comment string      // comment of attribute
+	// IsHide whether the attribute is hidden or not
+	IsHide bool
```
Comment (yingfeng) [inline pkg/vm/engine/types.go:44]: IsHide => IsHidden

**Comment by nnsgmsone** [comment]
> My eyes! So all it does, is to add a ctx parameter, is this true?

replace engine.Snapshot with client.TxnOperator

**Review by yingfeng — CHANGES_REQUESTED** [review]

File: pkg/vm/engine/types.go (hunk 1 above)
Comment (nnsgmsone) [reply pkg/vm/engine/types.go:44]: ok

**Review by yingfeng — APPROVED** [review]

**Comment by fengttt** [comment]
> 
> > My eyes! So all it does, is to add a ctx parameter, is this true?
> 
> replace engine.Snapshot with client.TxnOperator

I approved this PR.   However, this kind of change/reason etc, really should be in your commit message as well as PR decription.

**Review by fengttt — APPROVED** [review]

**Comment by nnsgmsone** [comment]
> I approved this PR. However, this kind of change/reason etc, really should be in your commit message as well as PR decription.

ok

**Comment by yingfeng** [comment]
Fix sca

**Comment by nnsgmsone** [comment]
> Fix sca

fixed

**Pull request merged**